Purchases/Purchasing/TransactionsFactory.swift

```diff
@@ -14,17 +14,41 @@
 
 import Foundation
 
-class TransactionsFactory {
-
-    func nonSubscriptionTransactions(withSubscriptionsData subscriptionsData: [String: [[String: Any]]],
-                                     dateFormatter: DateFormatterType) -> [Transaction] {
-        subscriptionsData.flatMap { (productId: String, transactionData: [[String: Any]]) -> [Transaction] in
-            transactionData.map {
-                Transaction(with: $0, productId: productId, dateFormatter: dateFormatter)
-            }.compactMap { $0 }
-        }.sorted {
-            $0.purchaseDate < $1.purchaseDate
-        }
+enum TransactionsFactory {
+
+    static func nonSubscriptionTransactions(withSubscriptionsData subscriptionsData: [String: [[String: Any]]],
+                                            dateFormatter: DateFormatterType) -> [StoreTransaction] {
+        subscriptionsData
+            .flatMap { (productId: String, transactionData: [[String: Any]]) -> [StoreTransaction] in
+                transactionData
+                    .map { SimpleTransaction(with: $0, productID: productId, dateFormatter: dateFormatter) }
+                    .compactMap { $0 }
+                    .map { StoreTransaction($0) }
+            }
+            .sorted { $0.purchaseDate < $1.purchaseDate }
+    }
+
+}
+
+/// `StoreTransactionType` backed by data parsed from the server
+private struct SimpleTransaction: StoreTransactionType {
```

Comment:
kinda feel like `SimpleTransaction` doesn't tell us much about what this is. I.e.: why is it simple, are the other ones complex? What's special about it? 

I think naming it `BackendTransaction` might be a bit more self-explanatory, or even `BackendParsedTransaction`

Reply:
I like that much better 👍